src/Microsoft.AspNet.Razor/Generator/RazorCodeGenerator.cs

```diff
@@ -11,20 +12,16 @@ public abstract class RazorCodeGenerator : ParserVisitor
     {
         private CodeGeneratorContext _context;
 
-        protected RazorCodeGenerator(string className, string rootNamespaceName, string sourceFileName, RazorEngineHost host)
+        protected RazorCodeGenerator(
+            string className,
+            [NotNull] string rootNamespaceName,
+            string sourceFileName,
+            [NotNull] RazorEngineHost host)
         {
             if (string.IsNullOrEmpty(className))
             {
                 throw new ArgumentException(CommonResources.Argument_Cannot_Be_Null_Or_Empty, "className");
```

Comment:
:-1: I'm keeping that separate since @hishamco has #374 already. that can be rebased on top of this and covers many more magic strings.